argus/util.py

```
@@ -362,3 +368,55 @@ def get_command(command, command_type=None):
         True: WINDOWS_NANO
     }
 }
+
+def get_os_type(instance):
+    """Get the OS type from a `Recipe` or `TestCase` instance."""
+    return instance._backend.remote_client.manager.os_type
+
+
+def skip_on_os(os_type):
+    """Skip the method decorated if we are using the `os_type`.
+    
+    :para func: Function to decorate.
+    :param os_type: Target OS.
+    """
+
+    # TODO(mmicu): A function based decorator with arguments is messy
+    # use a class based one
+    def decorator(func):
+
+        @six.wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                self = kwargs.get("self", args[0])
+            except IndexError:
+                LOG.error("Incompatible use of decorator %s on %s",
+                        "skip_on_os", func.__name__)
+                return func(*args, **kwargs)
+
+            instance_os_type = None
+            if isinstance(self, (base_recipe.BaseRecipe,
+                                base_testcase.BaseTestCase)):
+                try:
+                    instance_os_type = get_os_type(self)
+                except AttributeError as ex:
+                    raise exceptions.ArgusLogicError(
+                        "The OS type has not been determined yet in %s",
```

Maybe you should throw an error rather then ignoring.